Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -390,70 +432,63 @@ void CGPathAddPath(CGMutablePathRef path, const CGAffineTransform* m, CGPathRef
 
     if (pathObj->_count + copyObj->_count >= pathObj->_max) {
         pathObj->_max += copyObj->_count;
-        pathObj->_components = (pathComponent*)IwRealloc(pathObj->_components, pathObj->_max * sizeof(pathComponent));
+        pathObj->_elements = (CGPathElementInternal*)IwRealloc(pathObj->_elements, pathObj->_max * sizeof(CGPathElementInternal));
     }
 
     for (unsigned i = 0; i < copyObj->_count; i++) {
-        pathComponent c = copyObj->_components[i];
-
+        CGPathElementInternal c = copyObj->_elements[i];
```

Comment:
nit: no need to make a copy here.  You could apply the transform after you call pathObj->_elements[...] = copyObj->_elements[i];